File: BookingApp/src/main/java/com/example/BookingApp/reservations/model/RentingItemAvailability.java
package com.example.BookingApp.reservations.model;

import com.example.BookingApp.renting.model.RentingItem;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.util.Date;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Table(name="rentingItemAvailability")
public class RentingItemAvailability {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    @ManyToOne(fetch = FetchType.LAZY,cascade = CascadeType.ALL)
    private RentingItem rentingItem;
    private Date startTime;
    private Date endTime;
    private Double price;
}

File: BookingApp/frontend/src/app/shared/models/reservations/RentingItemAvailability.ts
import { RentingItem } from "./RentingItem";

export class RentingItemAvailability{
    public id : Number;
    public rentingItemId: Number;
    public startTime: Date;
    public endTime: Date;
    public price: Number;
    public rentingItem : RentingItem;

    constructor(id:Number,rentingItemId:Number,startTime : Date,endTime : Date,price:Number){
        this.id = id;
        this.rentingItemId = rentingItemId;
        this.startTime = startTime;
        this.endTime = endTime;
        this.price = price;
    }
    
}